pkg/registration/hub/gc/gc_resources.go

```diff
@@ -0,0 +1,62 @@
+package gc
+
+import (
+	"context"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	operatorhelpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/metadata"
+
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+)
+
+type resourceFilter struct {
+	gvr           schema.GroupVersionResource
+	labelSelector string
+}
+
+type gcResourcesController struct {
+	metadataClient      metadata.Interface
+	deletedResourceList []resourceFilter
+	eventRecorder       events.Recorder
+}
+
+func newGCResourcesController(
+	metadataClient metadata.Interface,
+	deletedResourceList []resourceFilter,
+	eventRecorder events.Recorder,
+) gcReconciler {
+	return &gcResourcesController{
+		metadataClient:      metadataClient,
+		deletedResourceList: deletedResourceList,
+		eventRecorder:       eventRecorder.WithComponentSuffix("gc-resources"),
+	}
+}
+
+func (r *gcResourcesController) reconcile(ctx context.Context, clusterName string, cluster *clusterv1.ManagedCluster) (gcReconcileOp, error) {
+	var errs []error
+	if cluster != nil && cluster.DeletionTimestamp.IsZero() {
+		return gcReconcileContinue, nil
+	}
+
+	background := metav1.DeletePropagationBackground
+	opts := metav1.DeleteOptions{PropagationPolicy: &background}
+	for _, resource := range r.deletedResourceList {
+		listOptions := metav1.ListOptions{
+			LabelSelector: resource.labelSelector,
+		}
+		err := r.metadataClient.Resource(resource.gvr).Namespace(clusterName).DeleteCollection(ctx, opts, listOptions)
+		if err != nil && !errors.IsNotFound(err) {
+			errs = append(errs, err)
+		}
+	}
```

Comment:
I think you need to get after deletecollection, so we can know how much is left.

Reply:
added count the remaining resources